blaze-server/src/main/scala/org/http4s/server/blaze/Http1ServerStage.scala

```diff
@@ -118,36 +118,38 @@ class Http1ServerStage(service: HttpService,
     val rr = new StringWriter(512)
     rr << req.httpVersion << ' ' << resp.status.code << ' ' << resp.status.reason << "\r\n"
 
-    val respTransferCoding = Http1Stage.encodeHeaders(resp.headers, rr, true) // kind of tricky method returns Option[Transfer-Encoding]
+    Http1Stage.encodeHeaders(resp.headers, rr, true)
+
+    val respTransferCoding = `Transfer-Encoding`.from(resp.headers)
+    val lengthHeader = `Content-Length`.from(resp.headers)
     val respConn = Connection.from(resp.headers)
 
     // Need to decide which encoder and if to close on finish
     val closeOnFinish = respConn.map(_.hasClose).orElse {
                           Connection.from(req.headers).map(checkCloseConnection(_, rr))
-                        }.getOrElse(parser.minorVersion() == 0)   // Finally, if nobody specifies, http 1.0 defaults to close
+                        }.getOrElse(parser.minorVersion == 0)   // Finally, if nobody specifies, http 1.0 defaults to close
 
     // choose a body encoder. Will add a Transfer-Encoding header if necessary
-    val lengthHeader = `Content-Length`.from(resp.headers)
-
     val bodyEncoder = {
-      if (req.method == Method.HEAD ||
-            (!resp.status.isEntityAllowed && lengthHeader.isEmpty && respTransferCoding.isEmpty)) {
+      if (req.method == Method.HEAD || !resp.status.isEntityAllowed) {
```

Comment:
I think we should log a warning if a body is detected with a response status that doesn't allow one since we are now cutting it out without the services consent.